### DISCUSSION TITLE
configure host tags (labels, groups)

### DISCUSSION THREAD

**Discussion by dpprdan** [twpayne/chezmoi#3502] [Q&A] [answered]
Does chezmoi support configuring host tags (a.k.a. labels, groups)?

Essentially I would like to do something like:

```toml
data:
    tags: [work, wsl]
```

```
{{ if in .tags "work" }}
# code for all hosts with the tag "work"
{{ end }}
{{ if in .tags "wsl" }}
# code for all hosts with the tag "wsl"
{{ end }}
{{ if (and (in .tags "work") (in .tags "wsl")) }}
# code for all hosts with the tags "work" and "wsl"
{{ end }}
```

I haven't been able to find anything to that effect in the docs (or issues/discussion).

A simple use case would be more than one "work" host. The docs mention in an example to use [hostname to capture machine-specific differences](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/#use-templates). But (IMHO) we often do not want to distiguish between specific machines or hosts but groups or types of hosts. E.g. `if eq .chezmoi.hostname "work-laptop"` will not work on a `new-work-laptop` that you want to provision, while still using `work-laptop`.

**Comment by twpayne** [comment, marked as answer]
I use [boolean flags for each tag](https://github.com/twpayne/dotfiles/blob/9dbfd6816b0ad23e85ac9cd369444eeb2f8e5a2a/home/.chezmoi.toml.tmpl#L1-L6).

If you want to use an array of tags then you can use the [`has` template function](https://masterminds.github.io/sprig/lists.html):

```
{{ if has "work" .tags }}
# code for all hosts with the tag "work"
{{ end }}
```

**Reply by dpprdan** [reply]
Awesome, thanks!

For those reading this and wondering what the syntax to negate a tag is:

```
{{ if not (has "work" .tags) }}
# code for all hosts without the tag "work"
{{ end }}
```

**Reply by dpprdan** [reply]
@twpayne Arrays aren't supported by any  `prompt[String|Choice]*` functions, are they?

**Reply by twpayne** [reply]
No. But if you want to read an array, you can read a string and then call [`splitList`](https://masterminds.github.io/sprig/string_slice.html).

**Reply by halostatue** [reply]
@dpprdan For what it's worth, we added [`promptMultichoice`](https://www.chezmoi.io/reference/templates/init-functions/promptMultichoice/){,Once} earlier this year.
